src/NServiceBus.Core/Serializers/Json/JsonMessageSerializer.cs

```diff
@@ -12,12 +12,11 @@ namespace NServiceBus.Serializers.Json
     using Newtonsoft.Json.Converters;
     using NServiceBus.Serialization;
 
-    /// <summary>
-    /// JSON message serializer.
-    /// </summary>
-    public class JsonMessageSerializer : IMessageSerializer
+    class JsonMessageSerializer : IMessageSerializer
```

Comment:
I think we need a way to obsolete the previously public types. Because if a user was using the public serializer directly and updated v6 this type is just gone now without obsolete. Maybe add an empty public class with obsolete and rename the internal serializer impl?